copilot.go

```diff
@@ -2,18 +2,148 @@ package main
 
 import (
 	"encoding/json"
+	"fmt"
+	"net/http"
 	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"time"
 )
 
-func getCopilotAuthToken() (string, error) {
-	// TODO: Windows?
-	bts, err := os.ReadFile(os.Getenv("HOME") + "/.config/github-copilot/hosts.json")
+const (
+	CopilotChatAuthURL   = "https://api.github.com/copilot_internal/v2/token"
+	CopilotEditorVersion = "vscode/1.95.3"
+	CopilotUserAgent     = "curl/7.81.0" // Necessay to bypass the user-agent check
+)
+
+type CopilotAccessToken struct {
+	Token     string `json:"token"`
+	ExpiresAt int64  `json:"expires_at"`
+	Endpoints struct {
+		API           string `json:"api"` // Can change in Github Enterprise instances
+		OriginTracker string `json:"origin-tracker"`
+		Proxy         string `json:"proxy"`
+		Telemetry     string `json:"telemetry"`
+	} `json:"endpoints"`
+	ErrorDetails *struct {
+		URL            string `json:"url,omitempty"`
+		Message        string `json:"message,omitempty"`
+		Title          string `json:"title,omitempty"`
+		NotificationID string `json:"notification_id,omitempty"`
+	} `json:"error_details,omitempty"`
+}
+
+type CopilotHTTPClient struct {
```

Comment:
```suggestion
type copilotHTTPClient struct {
```

don't need to export the struct I think